_posts/2017-11-09-containerised-application-deployment-solutions.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: Deployment solutions of a containerised application
+date: 2017-11-07
+---
+
+I have spent part of last week losing hairs over container deployments. If you have read my previous posts, you know by now that I have developed a very simple application and packaged into a Docker container. I am currently trying to deploy this application and I never thought deployment would be the challenging part. I looked at container deployments offered by cloud providers because I thought it would be the easiest and cheapest solutions. It probably is the cheapest but not necessarily the easiest!
+
+In this post I am sharing my experience with four cloud providers that offer container deployment service. If you are reading this article, bear in mind that I am no expert in cloud deployments and I am looking at a very particular use case: deployment of a Docker container that needs good availability (it is okay if it goes down once in a while) and does not need to be scalable (one instance running is enough).
+
+
+#### Microsoft Azure
+
+[Microsoft Azure](https://azure.microsoft.com/) offers very good [accounts](https://azure.microsoft.com/en-gb/free/free-account-faq/) that allow users to try many different services for free.
+In terms of container deployments it provides a wide range of solutions. It makes things a bit confusing to be honest:
+
+* [Azure Container Service](https://azure.microsoft.com/en-gb/services/container-service/) (AKS)
+<br/>This service seems to be great for complex containerised applications that require load-balancing, easy scaling, availability and container orchestration.
+<br/>In my case, while I want my service to be always available, the rest is not going to be a problem and the complexity this service seems to be adding is not worth it.
```

Comment:
Are you able to explain how the complexity affects you as a user? Is it hard to use or hard to understand with too much magic?

Reply:
I knew this one was coming...
I added a short sentence, I am trying to keep the length of the article short, tricky!